src/Test/Perf/infra/fetch_build.csx

```diff
@@ -4,39 +4,130 @@
+        var perfRunStatusPath = Path.Combine((Environment.ExpandEnvironmentVariables("%USERPROFILE%")), "PerfRunStatus.txt");
+        if (File.Exists(perfRunStatusPath))
+        {
+            lastBuild = File.ReadAllText(perfRunStatusPath);
+        }
+
+        StartWatching(sourceFolder, destinationFolder, lastBuild, perfRunStatusPath);
+
+        // Wait for 10 minutes before looking for new build again
+        Console.WriteLine($"Waiting for 10 minutes from {DateTime.Now} to look for new builds");
+        Thread.Sleep(10 * 60 * 1000);
+        Console.WriteLine($"Resume looking for new builds in {sourceFolder}");
+    }
+    // Dont let any exception to stop our Perf Automation
+    catch (System.Exception e)
+    {
+        var logFilePath = Path.Combine((Environment.ExpandEnvironmentVariables("%USERPROFILE%")), "PerfRunErrorLog.txt");
+        File.AppendAllLines(logFilePath, new[] { DateTime.Now.ToString()});
+        File.AppendAllLines(logFilePath, new[] { e.Message});
+        File.AppendAllLines(logFilePath, new[] { e.StackTrace});
+
+        CleanupSystem();
     }
 }
 
-if (latestBuild == null)
+void CleanupSystem()
 {
-    throw new InvalidOperationException($"Could not locate build with passing tests at location \"{sourceFolder}\".");
-}
+    var processNamesToKill = new[] {"csc", "vbc", "vbcscompiler", "devenv", "cpc" };
+    System.Diagnostics.Process.GetProcesses()
+        .Where(x => processNamesToKill.Any(p => x.ProcessName.ToLower().StartsWith(p)))
+        .ToList()
+        .ForEach(x => x.Kill());
 
-var latestBuildFolder = Path.Combine(sourceFolder, latestBuild);
+    //Cleanup the registry entries by CPC to make it ready for the next run 
+    var processInfo = new ProcessStartInfo(Environment.ExpandEnvironmentVariables(@"%SYSTEMDRIVE%\CPC\CPC.exe"));
+    processInfo.Arguments = @"/DisableArchive /Cleanup";
+    var process = Process.Start(processInfo);
 
-if (Directory.Exists(destinationFolder))
-{
-    Directory.Delete(destinationFolder, recursive: true);
+    // Shouldn't take more than 1 minute to cleanup
+    if(!process.WaitForExit(60000))
+    {
+        // If the CPC doesn't cleanup in a minute then we need to stop the automation and check the machine
+        var logFilePath = Path.Combine((Environment.ExpandEnvironmentVariables("%USERPROFILE%")), "PerfRunErrorLog.txt");
+        File.AppendAllLines(logFilePath, new[] { DateTime.Now.ToString()});
+        File.AppendAllLines(logFilePath, new[] { "CPC did not get cleaned up within 1 minute. Hence shutting down automation" });
+
+        Process.GetCurrentProcess().Kill();
+    }
 }
 
-CopyDirectory(latestBuildFolder, destinationFolder);
+void StartWatching(
+    string sourceFolder,
+    string destinationFolder,
+    string lastBuild,
+    string perfRunStatusPath)
+{
+    var lastBuildVersion = lastBuild == null ? null : Version.Parse(lastBuild);
+    foreach (var folder in Directory.GetDirectories(sourceFolder, "????????.?").Reverse())
```

Comment:
Oh, never mind, I saw your break at the end. In that case, this doesn't really need to be a loop since it only runs once?